### ISSUE TITLE
Restricting one user per notebook session

### ISSUE THREAD

**Issue by dvigneshwer** [jupyter/notebook#1777]
When multiple user agent access the notebook session the modification made by the different user agent is affecting the session started as it is accessing the same kernel queue, PFB the screenshot

![image](https://cloud.githubusercontent.com/assets/5075497/18709588/84f4678e-801e-11e6-90c2-f78bd45966ff.png)

Is there any way to restrict one user per session or have a seperate ZMQ queue started to for handling the modification of the sesond user agent

**Comment by takluyver** [comment]
Until we have real-time collaboration, it doesn't make much sense to have two users actively using the same notebook at the same time - they will be editing independent copies, and overwriting each other on every save (or getting a warning about the file changing on disk). For now, it makes more sense for each user to be working on a separate notebook.

**Comment by dvigneshwer** [comment]
I totally agree to the part of multiple users using an notebook session , think of environments which has multiple user agent using the same creditials for using the notebook application.

Consider cases where I have a notebook session spawned using the notebook api's in a webapp by embedding notebook application in an iframe and my webapp wants only one person to edit the notebook at a time for uniformity reason.

The only solution I see is to have multiple queue connected to the kernel for serving multiple user agents modification so that it does not affect the other users operations. Atlast some version controlling need to be done in order to keep the latest version and the other solution I see is like to have jupyterhub setup for spawning multiple notebook servers for multiplte user agent.

**Comment by ellisonbg** [comment]
Tagging this with the JupyterLab milestone as that is likely where this will end up fixed.
